### PR TITLE
update Canvas Clock

### PR DESCRIPTION
**Pull request by cherifGsoul**
This updates the Canvas Clock to use `can5` and `codepen`.

### CODE REVIEW

File: docs/can-guides/commitment/recipes/canvas-clock/canvas-clock.md [hunk 1]
```diff
@@ -14,9 +14,140 @@ In this guide you will learn how to:
 
 The final widget looks like:
 
-<a class="jsbin-embed" href="https://jsbin.com/jeligek/3/embed?output">
-  CanJS Canvas Clock on jsbin.com
-</a>
+```html
+<style>
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/canvas-clock/canvas-clock.md]: I think we should still use CodePen to embed the starter and finished code examples, like we do in this guide: https://canjs.com/doc/guides/recipes/signup-simple.html

File: docs/can-guides/commitment/recipes/canvas-clock/canvas-clock.md [hunk 2]
```diff
@@ -27,22 +158,61 @@ The following sections are broken down into the following parts:
...
+    <digital-clock time:from="time"/>
+    <analog-clock time:from="time"/>
+  `
+});
+</script>
+```
+@codepen
 
-This JS Bin has initial prototype HTML, CSS, and JS to bootstrap a basic CanJS application.
+This codepen has initial prototype HTML, CSS, and JS to bootstrap a basic CanJS application.
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/canvas-clock/canvas-clock.md]: “codepen” -> “CodePen” everywhere

File: docs/can-guides/commitment/recipes/canvas-clock/canvas-clock.md [hunk 3]
```diff
@@ -27,22 +158,61 @@ The following sections are broken down into the following parts:
...
 ## Setup ##
 
-__START THIS TUTORIAL BY CLONING THE FOLLOWING JS BIN__:
-
-> Click the **JS Bin** button.  The JS Bin will open in a new window. In that new window, under `File`, click `Clone`.
-
-<a class="jsbin-embed" href="https://jsbin.com/jeligek/1/embed?html,css,js,output">
-  CanJS Canvas Clock on jsbin.com
-</a>
+__START THIS TUTORIAL BY CLONING THE FOLLOWING CODEPEN__:
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/canvas-clock/canvas-clock.md]: This should match the “START THIS TUTORIAL BY CLICKING” text shown in this guide: http://localhost:8080/doc/guides/recipes/signup-simple.html

File: docs/can-guides/commitment/recipes/canvas-clock/6-min-hour-hands.js [hunk 4]
```diff
@@ -1,8 +1,10 @@
+import { Component } from "can";
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/canvas-clock/6-min-hour-hands.js]: I think these will need to be changed to `import { Component } from "//unpkg.com/can@5/core.mjs";`

Comment (cherifGsoul) [reply docs/can-guides/commitment/recipes/canvas-clock/6-min-hour-hands.js]: Done!

File: docs/can-guides/commitment/recipes/canvas-clock/1-setup.js [hunk 5]
```diff
@@ -1,21 +1,18 @@
...
-        const intervalID = setInterval(() => {
-          resolve( new Date() );
-        }, 1000);
-
-        resolve( new Date() );
-
-        return () => clearInterval(intervalID);
-      }
+    time: {Default: Date, Type: Date},
+    coonectedCallback() {
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/canvas-clock/1-setup.js]: coonectedCallback -> connectedCallback

File: docs/can-guides/commitment/recipes/canvas-clock/3-circle.js [hunk 6]
```diff
@@ -1,24 +1,33 @@
-can.Component.extend({
+import { Component } from "//unpkg.com/can@5/core.mjs";
+
+Component.extend({
   tag: "analog-clock",
-  view: '<canvas id="analog" width="255" height="255"></canvas>',
+  view: '<canvas this:to="canvasElement" id="analog" width="255" height="255"></canvas>',
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/canvas-clock/3-circle.js]: Let’s use template literals to contain strings with quotes:

```js
view: `<canvas this:to="canvasElement" id="analog" width="255" height="255"></canvas>`,
```

There’s a few places where this‘ll need to be changed.

File: docs/can-guides/commitment/recipes/canvas-clock/4-second-hand.js [hunk 7]
```diff
@@ -1,34 +1,44 @@
...
+    get canvas() {
+      return this.canvasElement.getContext("2d");
+    },
+
     connectedCallback(element) {
-      const canvas = element.firstChild.getContext("2d");
       const diameter = 255;
       const radius = diameter/2 - 5;
-      const center = diameter/2;
+			const center = diameter/2;
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/canvas-clock/4-second-hand.js]: Indentation

File: docs/can-guides/commitment/recipes/canvas-clock/canvas-clock.md [hunk 8]
```diff
@@ -254,9 +253,9 @@ In this section, we will:
 Update the __JavaScript__ tab to:
 
 @sourceref ./4-second-hand.js
-@highlight 1-3,22-48,only
+@highlight 3-5,31-57,only
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/canvas-clock/canvas-clock.md]: I think this should be `@highlight 3-5,31-58,only`

File: docs/can-guides/commitment/recipes/canvas-clock/6-min-hour-hands.js [hunk 9]
```diff
@@ -1,29 +1,37 @@
...
-        canvas.beginPath();
-        canvas.moveTo(center, center);
-        canvas.lineTo(x, y);
-        canvas.closePath();
-        canvas.stroke();
-      };
-
-      // draw second hand
       this.listenTo("time", (ev, time) => {
         canvas.clearRect(0, 0, diameter, diameter);
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/canvas-clock/6-min-hour-hands.js]: This code assumes there’s a `canvas` variable, but there isn’t. I think we should probably use `this.canvas` everywhere.

File: docs/can-guides/commitment/recipes/canvas-clock/canvas-clock.md [hunk 10]
```diff
@@ -326,14 +319,16 @@ You know everything at this point.  You got this!
 Update the __JavaScript__ tab to:
 
 @sourceref ./6-min-hour-hands.js
-@highlight 48-69,only
+@highlight 50-71,only
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/canvas-clock/canvas-clock.md]: I think the highlighting is off here.

Comment (cherifGsoul) [reply docs/can-guides/commitment/recipes/canvas-clock/canvas-clock.md]: Done!

File: docs/can-guides/commitment/recipes/canvas-clock/6-min-hour-hands.js (hunk 9 above)
Comment (cherifGsoul) [reply docs/can-guides/commitment/recipes/canvas-clock/6-min-hour-hands.js]: Done!

File: docs/can-guides/commitment/recipes/canvas-clock/canvas-clock.md (hunk 8 above)
Comment (cherifGsoul) [reply docs/can-guides/commitment/recipes/canvas-clock/canvas-clock.md]: Done!

File: docs/can-guides/commitment/recipes/canvas-clock/4-second-hand.js (hunk 7 above)
Comment (cherifGsoul) [reply docs/can-guides/commitment/recipes/canvas-clock/4-second-hand.js]: Done!

File: docs/can-guides/commitment/recipes/canvas-clock/3-circle.js (hunk 6 above)
Comment (cherifGsoul) [reply docs/can-guides/commitment/recipes/canvas-clock/3-circle.js]: Done!

File: docs/can-guides/commitment/recipes/canvas-clock/1-setup.js (hunk 5 above)
Comment (cherifGsoul) [reply docs/can-guides/commitment/recipes/canvas-clock/1-setup.js]: Done!

File: docs/can-guides/commitment/recipes/canvas-clock/4-second-hand.js [hunk 11]
```diff
@@ -1,34 +1,44 @@
...
-        canvas.closePath();
-        canvas.stroke();
+        this.canvas.lineWidth = 4.0;
+        this.canvas.strokeStyle = "#567";
+        this.canvas.beginPath();
+        this.canvas.arc(center, center, radius, 0, Math.PI * 2, true);
+        this.canvas.closePath();
+        this.canvas.stroke();
 
         Object.assign(canvas, {
```
Comment (chasenlehara) [inline docs/can-guides/commitment/recipes/canvas-clock/4-second-hand.js]: This should be `Object.assign(this.canvas, {`

Going through the guide from the starter CodePen and copying the highlighted code in each step should reveal any issues like this.

Comment (cherifGsoul) [reply docs/can-guides/commitment/recipes/canvas-clock/4-second-hand.js]: Fixed!

**Pull request merged**